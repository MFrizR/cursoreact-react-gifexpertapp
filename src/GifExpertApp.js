// Sniper rafc

import React, {useState} from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

export const GifExpertApp = () => {
    
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories,setCategories] = useState(['Dragon Ball'])

   /*    const handleAdd = () =>{
         //Agregar un elemento al arreglo.
         //setCategories('[HunterXHunter]',...categories);
          setCategories( cats => [...cats, 'Hunter']);
     } */

    return (
        <div>
            <h2>GifExpertApp</h2>
            {/* Renderizando el componente: */}
            <AddCategory setCategories= { setCategories }/>
            <hr />
           {/*  <button onClick={ handleAdd }>Agregar</button> */}
            <ol> 
               {
                   categories.map( category => (
                       /* return <li key={category}>{category}</li> */
                       <GifGrid 
                          
                            category= {category}
                            key= {category}
                       />)
                   )
               }
            </ol>

        </div>
    )
}
