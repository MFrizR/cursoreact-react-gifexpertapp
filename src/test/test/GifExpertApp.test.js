import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../../GifExpertApp';

describe('Pruebas en el GifExpertApp', () => {
    
    test('1.- El componente se debe mostrar correctamente ', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('2.- Debe de mostrar una lista de categorias ', () => {
        
        const categories = ['dragon','vegeta'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>);
        expect( wrapper ).toMatchSnapshot();
       // expect( wrapper.find('GifGrid').length ).toBe( categories.length);
       expect( wrapper.find('GifGrid').length ).toBe(1);
    })
    
})
