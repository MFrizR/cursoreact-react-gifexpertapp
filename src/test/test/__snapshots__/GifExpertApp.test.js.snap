// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifExpertApp 1.- El componente se debe mostrar correctamente  1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <Component
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Pruebas en el GifExpertApp 2.- Debe de mostrar una lista de categorias  1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <Component
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
